packages/flutter_tools/lib/src/vmservice.dart

```diff
@@ -417,14 +417,33 @@ extension FlutterVmService on vm_service.VmService {
 
   Uri get httpAddress => this != null ? _httpAddressExpando[this] : null;
 
+  Future<vm_service.Response> callMethodWrapper(
+    String method, {
+    String isolateId,
+    Map<String, dynamic> args
+  }) async {
+    try {
+      return await callMethod(method, isolateId: isolateId, args: args);
+    } on vm_service.RPCError catch(e) {
+      // If the service disappears mid-request the tool is unable to recover
+      // and should begin to shutdown due to the service connection closing.
+      // Swallow the exception here and let the shutdown logic elsewhere deal
+      // with cleaning up.
+      if (e.code == RPCErrorCodes.kServiceDisappeared) {
```

Comment:
Is this the only error code that needs this treatment?

Reply:
I believe so. We handle `kMethodNotFound` for extensions registered over the service protocol, but I don't think that's something we want to handle here since these extensions should be available as soon as the application starts up.